docs/docsite/rst/inventory/implicit_localhost.rst

```diff
@@ -0,0 +1,34 @@
+.. _implicit_localhost:
+
+Implicit 'localhost'
+====================
+
+When you try to use a 'localhost' and you don't have it defined in inventory, Ansible will create an implicit one for you.::
+
+    - hosts: all
+      tasks:
+        - name: check that i have log file for all hosts on my local machine
+          stat: path=/var/log/hosts/{{inventory_hostname}}.log
+          delegate_to: localhost
+
+In a case like this (or ``local_action``) Ansible needs to contact a 'localhost' but you did not supply one,
+since it is common enough and we 'should know' the settings, we create one for you.
+This host is defined with specific connection vars equivalent to this in an inventory::
+
+   ...
+
+   hosts:
+     localhost:
+      vars:
+        ansible_connection: local
+        ansible_python_interpreter: "{{ansible_playbook_python}}"
+
+This ensures that the proper connection and Python are used to execute your tasks locally,
+you can of course override this and use any specific settings you want by specifying a 'localhost' type host in your inventory.
```

Comment:
I think I'd spell this out more in its own paragraph.

You can override the built-in implicit `localhost` by creating a `localhost` host entry in your inventory. At that point, all implicit `localhost` behaviors are ignored. `localhost` in inventory is treated just like any other host — `group_vars` and `host_vars` set in inventory will apply, including connection and Python settings. This will also affect `delegate_to` and `local_action`.